en/orm/retrieving-data-and-resultsets.rst

```diff
@@ -497,8 +497,9 @@ load fields off of contained associations, you can use ``autoFields()``::
 
     // Select id & title from articles, but all fields off of Users.
     $query->select(['id', 'title'])
-        ->contain(['Users'])
-        ->autoFields(true);
+        ->contain(['Users' => function($q) {
+			return $q->autoFields(true);
+		          }]);
```

Comment:
@markstory I wrote a bunch of tests to see how ``autoFields()`` reacts with various associations types : https://github.com/HavokInspiration/cakephp/commit/39c5269e84391ed7b3f3bfaa91d28ebf1447faf9

It seems @caBBAlainB is right if I understood the issue correctly and your observation regarding the fact that "autoFields value [might not be] copied off the query builder in all association types."